Framework/CurveFitting/test/Functions/FunctionQDependsTest.h

```diff
@@ -0,0 +1,163 @@
+#ifndef MANTID_CURVEFITTING_FUNCTIONQDEPENDSTEST_H
+#define MANTID_CURVEFITTING_FUNCTIONQDEPENDSTEST_H
+
+// Mantid Coding standars <http://www.mantidproject.org/Coding_Standards>
+
+// Mantid Headers from the same project
+#include "MantidCurveFitting/Functions/FunctionQDepends.h"
+// Mantid headers from other projects
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/ParamFunction.h"
+#include "MantidAPI/NumericAxis.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include "MantidKernel/EmptyValues.h"
+#include "MantidDataHandling/LoadNexus.h"
+// 3rd party library headers
+#include <cxxtest/TestSuite.h>
+// Standard library
+// N/A
+
+using Attr = Mantid::API::IFunction::Attribute;
+
+namespace {
+class ImplementsFunctionQDepends
+    : public Mantid::CurveFitting::Functions::FunctionQDepends {
+
+public:
+  std::string name() const override { return "ImplementsFunctionQDepends"; }
+
+  void function1D(double *out, const double *xValues,
+                  const size_t nData) const override {
+    double Q = this->getAttribute("Q").asDouble();
+    for (size_t i = 0; i < nData; i++) {
+      out[i] = Q * xValues[i];
+    }
+  }
+};
+} // end of namespace
+
+class FunctionQDependsTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static FunctionQDependsTest *createSuite() {
+    return new FunctionQDependsTest();
+  }
+  static void destroySuite(FunctionQDependsTest *suite) { delete suite; }
+
+  void testConstruction() {
+    TS_ASSERT_THROWS_NOTHING(ImplementsFunctionQDepends f);
+  }
+
+  void testSetWorkspace() {
+    double startX{0.0}, endX{1.0};
+    ImplementsFunctionQDepends f;
+    // test with an non matrix workspace
+    TS_ASSERT_THROWS_NOTHING(
+        f.setMatrixWorkspace(this->unsuitableWS(), 0, startX, endX));
```

Comment:
If the workspace is unsuitable, should this maybe throw an exception? Not sure if this situation can occur in the real world, but if it's completely unsuitable what is the function going to do?